### PR TITLE
Skip database reset on partial test run

### PR DESCRIPTION
**Pull request by yguedidi**

### CODE REVIEW

**Comment by yguedidi** [comment]
@j0k3r this should support your use case of partial test run ;)

File: tests/functions.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+function isPartialRun(): bool
+{
+    foreach ($_SERVER['argv'] as $arg) {
+        if (str_starts_with($arg, '--filter')) {
```
Comment (yguedidi) [inline tests/functions.php:6]: used `str_starts_with` to support the case of `--filter=...` instead of `--filter ...`

File: composer.json [hunk 2]
```diff
@@ -200,7 +200,10 @@
     "autoload-dev": {
         "psr-4": {
             "Tests\\": "tests/"
-        }
+        },
+        "files": [
+            "tests/functions.php"
```
Comment (yguedidi) [inline composer.json:205]: I preferred a separate file than having the function definition in the bootstrap file

File: tests/functions.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+function isPartialRun(): bool
```
Comment (j0k3r) [inline tests/functions.php:3]: Could you add a comment above to explain the goal of the file and the function?

**Review by j0k3r — COMMENTED** [review]
Overall that's a good idea/workaround 👍🏼

File: tests/functions.php (hunk 3 above)
Comment (yguedidi) [reply tests/functions.php:3]: sure!

Comment (yguedidi) [reply tests/functions.php:11]: done

**Review by j0k3r — APPROVED** [review]

**Pull request merged**